src/parser/statement.js

```diff
@@ -530,12 +530,11 @@ pp.parseFor = function (node, init) {
 // same from parser's perspective.
 
 pp.parseForIn = function (node, init, forAwait) {
-  let type;
+  const type = this.match(tt._in) ? "ForInStatement" : "ForOfStatement";
   if (forAwait) {
+    node.await = true;
     this.eatContextual("of");
-    type = "ForAwaitStatement";
   } else {
```

Comment:
Using `Function` as a reference, the `generator` and `async` flags are explicitly set to `false` when they don't apply, so it seems like it should do the same here.